authentication/src/main/java/io/camunda/authentication/CamundaOAuthPrincipalService.java

```diff
@@ -74,58 +81,74 @@ public CamundaOAuthPrincipalService(
 
   public OAuthContext loadOAuthContext(final Map<String, Object> claims)
       throws OAuth2AuthenticationException {
+    final var authContextBuilder = new AuthenticationContextBuilder();
+    final var principals = oidcPrincipalLoader.load(claims);
+    final var username = principals.username();
+    final var clientId = principals.clientId();
+
+    final var principalIdentifiers = new HashMap<EntityType, Set<String>>();
+
+    if (username == null && clientId == null) {
+      throw new OAuth2AuthenticationException(
+          new OAuth2Error(OAuth2ErrorCodes.INVALID_CLIENT),
+          "Neither username claim (%s) nor clientId claim (%s) could be found in the claims. Please check your OIDC configuration."
+              .formatted(usernameClaim, clientIdClaim));
+    }
```

Comment:
❓ Must users always configure a username claim or client id claim? What if they only want to use mapping-rules and no simple mappings?